AWSCognitoIdentityProvider/AWSCognitoIdentityUser.m

```diff
@@ -203,13 +229,28 @@ -(instancetype) initWithUsername: (NSString *)username pool:(AWSCognitoIdentityU
         self.confirmedStatus = AWSCognitoIdentityUserStatusConfirmed;
 
         NSString * accessTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoIdentityUserAccessToken];
+        NSString * idTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoIdentityUserIdToken];
+        
+        NSString * idToken = self.pool.keychain[idTokenKey];
         NSString * accessToken = self.pool.keychain[accessTokenKey];
-        //if the session expires > 5 minutes return it and there is at least an accessToken.
-        if(expiration && [expiration compare:[NSDate dateWithTimeIntervalSinceNow:5 * 60]] == NSOrderedDescending && accessToken){
-            NSString * idTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoIdentityUserIdToken];
-            AWSCognitoIdentityUserSession * session = [[AWSCognitoIdentityUserSession alloc] initWithIdToken:self.pool.keychain[idTokenKey] accessToken:accessToken refreshToken:refreshToken expirationTime:expiration];
         
-            session.expirationTime = expiration;
+        AWSCognitoIdentityUserSession * session;
+        
+        // Session is available if we have expiration and accessToken.
```

Comment:
I don't know the API, but you're passing refresh token without checking it first. Any cases where that's going to bite us?

Reply:
I think it should be okay, AWSCognitoIdentityUserSession takes a null refresh token.